### ISSUE TITLE
Retry on error documentation example does not work

### ISSUE THREAD

**Issue by Atharex** [argoproj/argo#2065]
**What happened**:
The example from the official docs returns a unknown field "retryPolicy" error.
https://github.com/argoproj/argo/blob/master/examples/retry-on-error.yaml

**What you expected to happen**:
The example to run normally when a retryPolicy field is set.

**How to reproduce it (as minimally and precisely as possible)**:
Just try to run the example from the official docs

**Anything else we need to know?**:
The example works fine if the "retryPolicy" field is omitted.
Might be a versions issue. For which argo versions are those docs written?

**Environment**:
- Argo version:
```
$ argo version
argo: v2.4.3
  BuildDate: 2019-12-06T03:36:01Z
  GitCommit: cfe5f377bc3552fba90afe6db7a76edd92c753cd
  GitTreeState: clean
  GitTag: v2.4.3
  GoVersion: go1.11.5
  Compiler: gc
  Platform: linux/amd64
```
- Kubernetes version :
```
$ kubectl version -o yaml
clientVersion:
  buildDate: "2020-01-18T23:30:10Z"
  compiler: gc
  gitCommit: 59603c6e503c87169aea6106f57b9f242f64df89
  gitTreeState: clean
  gitVersion: v1.17.2
  goVersion: go1.13.5
  major: "1"
  minor: "17"
  platform: linux/amd64
serverVersion:
  buildDate: "2020-01-18T23:22:30Z"
  compiler: gc
  gitCommit: 59603c6e503c87169aea6106f57b9f242f64df89
  gitTreeState: clean
  gitVersion: v1.17.2
  goVersion: go1.13.5
  major: "1"
  minor: "17"
  platform: linux/amd64
```

---

**Message from the maintainers**:

If you are impacted by this bug please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

**Comment by simster7** [comment]
What is the exact error you're seeing here?

**Comment by Atharex** [comment]
> What is the exact error you're seeing here?

2020/01/27 14:58:37 Failed to parse workflow: error unmarshaling JSON: while decoding JSON: json: unknown field "retryPolicy"

**Comment by sarabala1979** [comment]
`retryPolicy`  included on 2.5-rc1 release. Try on the latest build `v2.5-rc3`.
Here the 2.4.x examples: https://github.com/argoproj/argo/tree/release-2.4/examples
